app/models.py

```diff
@@ -198,6 +198,42 @@ def nearest_neighbors(self, limit=10):
                 limit(limit)
 
     @property
+    def has_fully_registered(self):
+        '''
+        Returns whether the user has fully completed the registration/signup
+        flow.
+        '''
+
+        join_event = db.session.query(SimpleUserEvent).\
+                     filter(SimpleUserEvent.user_id==self.id).\
+                     filter(SimpleUserEvent.subtype==\
+                            SimpleUserEvent.SUBTYPE_USER_JOINED).\
+                     all()
+        return len(join_event) > 0
```

Comment:
Hmm, I'm trying that but I'm getting `TypeError: Boolean value of this clause is not defined`...